File: IRCLib/src/main/java/net/darkmeow/irc/data/DataUser.java
package net.darkmeow.irc.data;

import lombok.Getter;
import net.darkmeow.irc.data.enmus.EnumUserPremium;
import org.jetbrains.annotations.NotNull;

public class DataUser {

    @Getter
    @NotNull
    private final String name;

    @Getter
    @NotNull
    private final EnumUserPremium premium;

    @Getter
    private final DataUserState state;

    public DataUser(@NotNull String name, @NotNull EnumUserPremium premium, DataUserState state) {
        this.name = name;
        this.premium = premium;
        this.state = state;
        new DataSkin(new byte[] {}, new byte[] {}, false);
    }

}

File: IRCServer/src/main/kotlin/net/darkmeow/irc/data/base/DataUser.kt
package net.darkmeow.irc.data.base

import net.darkmeow.irc.data.enmus.EnumUserPremium
import org.mindrot.jbcrypt.BCrypt

/**
 * 用户数据
 *
 * @param name 用户名
 * @param metadata 元素据
 */
data class DataUser(
    val name: String,
    val metadata: UserMetadata,
) {
    data class UserMetadata(
        val premium: EnumUserPremium,
        private val passwordHash: String
    ) {
        /**
         * 检查密码是否正确
         *
         * @param password 客户端传输过来的密码
         */
        fun checkPassword(password: String) = BCrypt.checkpw(password, passwordHash)
    }
}